### ISSUE TITLE
parameterize number of cores in main.nf

### ISSUE THREAD

**Issue by nhoffman** [nhoffman/dada2-nf#89]
COU allocation is hard-coded for cmsearch and vsearch - either add parameters for these in the profile, or use as many are available within the task.

**Comment by dhoogest** [comment]
Scott pointed this out today as well when reviewing load on agrajag (I was running some NGS16S dev analysis there). Vsearch step took all available cores, which for that machine is a lot : )

**Comment by crosenth** [comment]
How does this look? https://github.com/nhoffman/dada2-nf/commit/594c6ecd187fb64c7cca9d736d30dc1cf9e3807f

**Comment by dhoogest** [comment]
Looks fine to me as a simple solution!

**Comment by dhoogest** [comment]
Does `cmsearch` also allow for ncores specification? Sounds like that step is also greedy

**Comment by crosenth** [comment]
Yes here or is it not working?  https://github.com/nhoffman/dada2-nf/blob/master/main.nf#L180

**Comment by crosenth** [comment]
Not sure what profile you using but queueSize should also be set to a reasonable number:

https://github.com/nhoffman/dada2-nf/blob/master/nextflow.config#L48

**Comment by crosenth** [comment]
See `--cpus `

- https://github.com/nhoffman/dada2-nf#execution
